com.unity.sg2/Editor/GraphDeltaRegistry/GraphDelta/GraphDelta.cs

```diff
@@ -135,6 +136,11 @@ public bool ReconcretizeNode(ElementID id, Registry registry)
                 return true;
             }
             var nodeHandler = m_data.GetHandler(id, this, registry).ToNodeHandler();
+            if (nodeHandler == null)
+            {
+                Debug.LogError("Failed to reconcretize node with ID: " + id.FullPath);
```

Comment:
Just a general question here. Should we be logging errors or throwing them and letting the thrown errors bubble up?

Reply:
I don't think we have a consistent method of handling errors/exceptions right now. 
I'm currently in the mindset that we log them so as to immediately draw attention to the area of failure as well as leave a little bit of a bookmark in the code that this error condition occurs, so it can be fixed up later

Also, not having this handling here can basically break a graph asset during a reimport (when an asset is saved) so this is also guarding against that case.

Reply:
In this case, if this error is hit, the node definition is invalid and the user can't take action. I think this is the case where an error should be handled at a lower level or shown in the UI.

Restated: I don't think that the user will need to understand "reconcretizing".

Reply:
Yea- Logs should be user actionable. I believe this ought to actually be an exception. If you try to reconcretize a node that doesn't exist or isn't found, that's equivalent to bad fetch from any normal data structure, which is typically an exception.